### ISSUE TITLE
Bug in ray transform

### ISSUE THREAD

**Issue by MatthiasJE** [odlgroup/odl#946]
I want to compute the ray transform for one angle. The following code shows that there is a bug. I suppose that there is an easy fix which treats one angle differently.

```python
import odl
import numpy as np

X = odl.uniform_discr(min_pt=[-1, -1], max_pt=[1, 1], shape=[100, 100])
angle_partition = odl.uniform_partition(0, np.pi, 1)
detector_partition = odl.uniform_partition(-2, 2, 170)
geometry = odl.tomo.Parallel2dGeometry(angle_partition, detector_partition)
ray_transform = odl.tomo.RayTransform(X, geometry)

ray_transform(X.one())
```

```
Traceback (most recent call last):

  File "<ipython-input-1-c32811242236>", line 10, in <module>
    ray_transform(X.one())

  File "odl/operator/operator.py", line 695, in __call__
    out = self._call_out_of_place(x, **kwargs)

  File "odl/tomo/operators/ray_trafo.py", line 226, in _call
    return scikit_radon_forward(x, self.geometry, self.range, out)

  File "odl/tomo/backends/scikit_radon.py", line 98, in scikit_radon_forward
    sinogram = scikit_range.element(radon(volume.asarray(), theta=theta).T)

  File "/home/ehrhardt/.local/lib/python2.7/site-packages/skimage/transform/radon_transform.py", line 91, in radon
    radon_image = np.zeros((padded_image.shape[0], len(theta)))

TypeError: object of type 'numpy.float64' has no len()
```

**Comment by aringh** [comment]
To me it looks like the problem is in the back-end, in `scikit`. I can for example run the following by using the `astra` back-end
```
>>> import numpy as np
>>> import odl
>>> reco_space = odl.uniform_discr([-1,-1], [1,1], [10,10])
>>> angle_partition = odl.uniform_partition(0, np.pi, 1)
>>> detector_partition = odl.uniform_partition(-2, 2, 100)
>>> geometry = odl.tomo.Parallel2dGeometry(angle_partition, detector_partition)
>>> ray_trafo = odl.tomo.RayTransform(reco_space, geometry, impl='astra_cuda')
>>> ray_trafo(reco_space.one())
uniform_discr([0.0, -2.0], [3.1415926535897931, 2.0], (1, 100), axis_labels=('$\\theta$', '$s$')).element([[0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0]])
```

**Comment by MatthiasJE** [comment]
I think the problem is in the way how ODL talks to `scikit`. You can run the scikit Radon transform perfectly well with 1 angle.

**Comment by aringh** [comment]
You are absolutely right :-) But you seem to have to give it to `scikit` as an array:
```
>>> my_image = np.array([[1,1,1], [1,1,1], [1,1,1]])
>>> skimage.transform.radon(my_image, [1])
array([[ 0.01744452],
       [ 2.98253971],
       [ 3.00060922],
       [ 2.98253971],
       [ 0.01744452]])
>>> skimage.transform.radon(my_image, 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/aringh/miniconda3/envs/odl-tvr-dart-py35/lib/python3.5/site-packages/skimage/transform/radon_transform.py", line 91, in radon
    radon_image = np.zeros((padded_image.shape[0], len(theta)))
TypeError: object of type 'int' has no len()
```

**Comment by MatthiasJE** [comment]
True. The question is now, how to change ODL so that this happens. I think the code I wrote in my first post should be perfectly legitimate.

**Comment by aringh** [comment]
Chaining line 98 in odl/tomo/backends/scikit_radon.py to
```
sinogram = scikit_range.element(radon(volume.asarray(), theta=[theta]).T)
```
i.e., adding the brackets around theta, seems to fix it for the 1-angle case. But it will cause a crash for the other cases. One could simply try to do a check of the form
```
if type(theta) is not list
```
on the line above, but I will await answer from @kohr-h or @adler-j first. They probably have a nicer way to solve the problem :-)

**Comment by adler-j** [comment]
It seems to me that the error is in the call `np.squeeze` [here](https://github.com/odlgroup/odl/blob/149e73ae7c9b1c3da495c96fac0b10f8599734f2/odl/tomo/backends/scikit_radon.py#L34):

```
return np.asarray(geometry.motion_grid).squeeze() * 180.0 / np.pi
```

clearly, if `geometry.motion_grid` is of length 1, this gives an error.

An easy fix is to change it to

```python
return geometry.angles * 180.0 / np.pi
```

would you like to make a PR with this change?

**Comment by adler-j** [comment]
Closed by #947.
